fix: Migrate all rules from eslint-plugin-twenty to eslint-plugin-twenty-ts

anikdhabal opened the pull request:
fixes #1590
fixes #1591 
fixes #1592 
fixes #1593

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/twentyhq/twenty?pullRequest=1618) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/twentyhq/twenty?pullRequest=1618) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/twentyhq/twenty?pullRequest=1618) it.</sub>

charlesBochet reviewed (commented):
Wow looks great! Can we completely remove the old linter plugin? (eslint-plugin-twenty)

anikdhabal commented:
> Wow looks great! Can we completely remove the old linter plugin? (eslint-plugin-twenty)

Yes we can remove.

charlesBochet commented on server/.eslintrc.js:
```diff
@@ -5,7 +5,7 @@ module.exports = {
     sourceType: 'module',
   },
-  plugins: ['@typescript-eslint/eslint-plugin', 'import', 'unused-imports'],
+  plugins: ['@typescript-eslint/eslint-plugin', 'import', 'unused-imports','twenty-ts'],
```
You'll need to add the eslint-plugin-twenty-ts to server package.json and to add its installation to server yarn install command (similar to what has been done in front/package.json)
I suspect this might be not straightforward + you'll need to fix many linter errors, so let's do that in another PR :)

anikdhabal replied:
I am going fix those now. Thanks for your review.

anikdhabal replied:
Yes, let's fix all the linters errors in another PR.

charlesBochet commented on packages/eslint-plugin-twenty-ts/src/rules/matching-state-variable.ts:
```diff
@@ -0,0 +1,100 @@
+import { Rule } from 'eslint';
```
could you use the same format as effect-component.ts to define these rules?

charlesBochet replied:
Being consistent across the codebase is one of our focus :)

anikdhabal replied:
OKK

anikdhabal replied:
@charlesBochet Done update the effect-component.ts and the test.

yashug replied:
@anikdhabal i think we need to follow the format from effect-component.ts, but not the other way around

charlesBochet replied:
yes exactly :) Also we don't use export default in the project :)

anikdhabal replied:
fixed

akhill10 commented on packages/eslint-plugin-twenty-ts/src/rules/no-hardcoded-colors.ts:
```diff
@@ -0,0 +1,26 @@
+import { Rule } from "eslint";
+
+const rule: Rule.RuleModule = {
```
`meta` values are missing in this rule.

anikdhabal replied:
Ahh, oversight, thank you.

charlesBochet commented on packages/eslint-plugin-twenty-ts/src/tests/all.spec.test.ts:
```diff
@@ -0,0 +1,90 @@
+import { RuleTester } from "@typescript-eslint/rule-tester";
```
let's rename this file effect-component.spec.ts

I've also notice that test files are inconsistently named. They should all be suffixed with .spec.ts :)

anikdhabal replied:
done

anikdhabal commented:
@charlesBochet can I delete all the old packages in this PR or make another PR for that along with some linter fix?

yashug commented:
@anikdhabal please also consider updating this file also with new ts rules https://github.com/twentyhq/twenty/blob/main/front/.eslintrc.js

anikdhabal commented:
> @anikdhabal please also consider updating this file also with new ts rules https://github.com/twentyhq/twenty/blob/main/front/.eslintrc.js

Done, thanks mate

anikdhabal commented:
@charlesBochet Fixed and Updated all related problem. Pls review once. Removed the old packages, fixed lint problem, followed the format from effect-component.ts.

charlesBochet reviewed (approved):
Merging it

anikdhabal commented:
> Merging it

Thanks a lot.

charlesBochet commented:
/tip $100 @anikdhabal

charlesBochet commented:
@anikdhabal I have used code from your PR and re-applied some from neo so, I will still split the bounty.
I'm in touch with Algora so this won't happen again :)

charlesBochet commented:
Regarding the PR, the tests of the packages are broken:

```
cd ./packages/eslint-plugin-twenty-ts 
yarn test
```

Could you take a look? If you can't that's fine, I can take care of them later :)

anikdhabal commented:
> Regarding the PR, the tests of the packages are broken:
> 
> ```
> cd ./packages/eslint-plugin-twenty-ts 
> yarn test
> ```
> 
> Could you take a look? If you can't that's fine, I can take care of them later :)

Why not, let me handle it.

anikdhabal commented:
> @anikdhabal I have used code from your PR and re-applied some from neo so, I will still split the bounty. I'm in touch with Algora so this won't happen again :)

Not an issue. Thank you again.

anikdhabal commented:
> /tip $100 @anikdhabal

How to claim this one? I've seen that it automatically added, but here it redirects me to the installation page.

charlesBochet commented:
It has been sent, thank you :)
